docs/public-networks/concepts/parallel-transaction-execution.md

```diff
@@ -0,0 +1,189 @@
+
+:::warning Important
+Parallel transaction execution is an early access feature.
+You can enable it using the `--Xbonsai-parallel-tx-processing-enabled` option.
+:::
+
+## Parallelization mechanism overview
+
+When parallel transaction execution is enabled, Besu initially executes all transactions within a
+block in parallel, operating under the optimistic assumption that they can all be executed
+concurrently without conflict.
+This parallel execution runs in the background, and Besu proceeds to sequentially process the
+transactions without waiting for the parallel execution to complete.
+
+The following flowchart outlines the sequential processing flow:
+
+<p align="center">
+
+```mermaid
+graph TD;
+  A(Start sequential processing) --> B{{Is transaction finalized?}};
+  B --> |Yes| C{{Conflict check}};
+  C --> |No conflict| D(Apply background state modifications);
+  C --> |Conflict detected| E(Replay transaction using background cache);
+  B --> |No| F(Execute transaction sequentially);
+  D --> G(End sequential processing);
+  E --> G;
+  F --> G;
+```
+
+</p>
+
+Besu first determines if a transaction has been "finalized," or completed by the background parallel
+execution:
+
+- **Finalized:** If the transaction is finalized, Besu examines whether there are any conflicts with
+  previously executed transactions.
+  - **No conflict:** If no conflict is detected, Besu directly applies the state modifications
+    generated in the background to the block, avoiding re-execution.
+  - **Conflict detected:** If a conflict is detected, Besu replays the transaction, using a cache of
+    background reads to improve efficiency.
+- **Not finalized:** If the transaction is not finalized, Besu executes it sequentially within the
+  block to ensure its completion, independent of the background execution.
+
+### Conflict detection strategy
+
+Besu's conflict detection strategy uses the *accumulator*, a
+[Bonsai Tries](data-storage-formats.md#bonsai-tries) feature that tracks addresses and slots touched
+or modified during a block's or transaction's execution.
+
+:::tip
+You can read more about Bonsai Tries in [Consensys' Guide to Bonsai Tries](https://consensys.io/blog/bonsai-tries-guide).
+:::
+
+If a slot, code, or anything else related to an account is modified, the Bonsai accumulator keeps
+track of this information.
+This strategy leverages Bonsai's storage benefits, only keeping track of block-to-block state diffs
+in Besu storage.
+
+The following flowchart outlines how Besu detects conflicts and imports transactions into the block:
+
+<p align="center">
+
+```mermaid
+graph TD;
+  A(Start block import) --> B(Fetch block's touched addresses);
+  B --> C{{For each transaction}};
+  C -->|Next transaction| D(Fetch transaction's touched addresses);
+  D --> E{{Compare addresses}};
+  E -->|Conflict detected| F(Replay transaction using prior values as cache);
```

Comment:
```suggestion
  E -->|Conflict detected| F(Replay transaction using cached data);
```
I may be wrong here, but feel this needs tweaking.

Reply:
Checking with @non-fungible-nelson.

Reply:
Latter is correct -- or cached values, differ to you all